src/lib/HsToCoq/ConvertHaskell/Module.hs

```diff
@@ -170,17 +171,17 @@ renameModule :: GlobalMonad r m => ModuleName -> m ModuleName
 renameModule mod = view $ edits.renamedModules.at mod.non mod
```

Comment:
It would definitely simply `convertModule`, but it would make `convertModules` more complicated, in particular this part on line 245:

`(((\(name,_,_) -> renameModule name) <&&&> pure . pure @NonEmpty))`

Here it seems we really only need the name of the module, but if we changed the type of `renameModule` we would have to provide the export list here.

Reply:
That makes sense. However, I notice that `convertModules` has the following type signature:
```Haskell
convertModules :: GlobalMonad r m => [(ModuleName, HsGroup GhcRn, [GHC.Name])] -> m [NonEmpty ConvertedModule]
```
Isn't ` [(ModuleName, HsGroup GhcRn, [GHC.Name])]` and `[(ModuleData, HsGroup GhcRn)]` equivalent? In that case, would it be possible to write the line of code you quoted just as before (`renameModule . fst <&&&> pure . pure @NonEmpty`), if the type of `renameModule` has been changed to work on `ModuleData`? Does it make sense, or I'm missing something?

Reply:
That makes sense, and I have tried making this change, but it seems that this has caused other issues to arise in `convertModules`. One of them has to do with `mergedModulesNELs` (line 243). Basically, this map used to be indexed by `ModuleName`, but after changing the type of `renameModule` and changing the above line to `(renameModule . fst <&&&> pure . pure @NonEmpty)`, this map is now indexed by `ModuleData`. This means that `ModuleData` now has to be an instance of `Ord`. Moreover, I might be misunderstanding something, but I don't think it makes sense to index this map by `ModuleData`. What would the export list of this `ModuleData` be? It seems that the purpose of this map is to link post-rename _names_ to information about modules that have that post-rename name, e.g. the combined list of exports from such modules. So it seems that the keys of this map should just be of type `ModuleName`, but after making the suggested changes, they are of type `ModuleData`. Does this make sense?

Reply:
That makes a lot of sense. I suppose changing the type of `renameModule` would not make things better then. Thanks a lot for trying this out!